opensafely/check.py

```diff
@@ -97,12 +101,14 @@ def get_repository_name():
     if "GITHUB_REPOSITORY" in os.environ:
         return os.environ["GITHUB_REPOSITORY"]
     else:
-
-        url = subprocess.run(
-            args=["git", "config", "--get", "remote.origin.url"],
-            capture_output=True,
-            text=True,
-        ).stdout
+        git_config_path = Path(".git", "config")
+        if not git_config_path.is_file():
+            return
+        config = configparser.ConfigParser()
+        config.read(git_config_path)
```

Comment:
thought 1: Let's say the file exists, but `config.read()` fails for some obscure reason (maybe the config file is locked by some other process). Then you'll simply get no remote in the same way as you would if the file didn't exist.

There are several different failure modes:

* no `.git/config`
* can't read `.git/config`
* no `remote "origin"` in `.git/config`

Maybe it's useful to distinguish these to the user, but it may well not be.